### ISSUE TITLE
Run Thrust unit tests with `__half`

### ISSUE THREAD

**Issue by brycelelbach** [NVIDIA/thrust#1408]
It seems we do not do this. It'd probably be a good idea. We could potentially add it to one of the list of tests types - maybe removing another type to make room for it. `__half` is a nice thing to test because it's numeric-like but not a builtin type.

**Comment by allisonvacanti** [comment]
Agreed. Just to note, I ran into issues trying to benchmark with `__half`, seems the thrust rng facilities don't support it. There may be quite a bit of infrastructure work involved in making this happen.

**Comment by griwes** [comment]
Note that this should probably also be extended to any other library floating point types we may be shipping.
